interview-questions.md

```diff
@@ -8,62 +8,67 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is iteration?
 
-  Your answer:
+  Your answer: How many times a function is repeated.
 
-  Researched answer:
+  Researched answer: Iteration is the process of executing a block of code over and over again until a condition is met.
 
 
 
 2. The built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: There has to be an array attatched to .map() in order for the function to work. 
+
+  Researched answer: The value of the array at the current iteration (required).
 
-  Researched answer:
+The index of the array at the current iteration (optional).
+
+The array the higher-order function was called on (optional).
 
 
 
 3. What is object destructuring?
 
-  Your answer:
+  Your answer: Assigns variables from an array by taking it apart.
 
-  Researched answer:
+  Researched answer: What's great about the destructuring assignment is that it allows you to take something like an array or an object and unpack their values into individual variables.
 
 
 
 4. What is the difference between an object and a class?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: An object is just a container for a name while a class is a type of function that includes objects and their properties.
 
+  Researched answer: Classes are a particular type of function that contain data and behavior. Classes are the blueprints of objects. Objects are collections of key:value pairs.
 
 
 5. Why would you use the method super()?
 
-  Your answer:
+  Your answer: when a child class is using the same object from a parent class, super() is used to use the information.
 
-  Researched answer:
+  Researched answer:  Within in the constructor method we call super() which passes the attributes from the constructor in the parent class
```

Comment:
yeah, they key here is calling the constructor() method from the parent class to build out it's this's.